vm/src/builtins/type.rs

```diff
@@ -523,6 +523,56 @@ impl PyType {
         Ok(())
     }
 
+    #[pygetset(magic)]
+    fn annotations(&self, vm: &VirtualMachine) -> PyResult<PyObjectRef> {
+        if !self.slots.flags.has_feature(PyTypeFlags::HEAPTYPE) {
+            return Err(vm.new_attribute_error(format!(
+                "type object '{}' has no attribute '__annotations__'",
+                self.name()
+            )));
+        }
+
+        let __annotations__ = identifier!(vm, __annotations__);
+        let annotations = self.attributes.read().get(__annotations__).cloned();
+
+        let annotations = annotations.unwrap_or_else(|| {
+            let annotations: PyObjectRef = vm.ctx.new_dict().into();
+            self.attributes
+                .write()
+                .insert(__annotations__, annotations.clone());
+            annotations
+        });
```

Comment:
I'd like to keep my original suggestion for a few (weak) reasons.
- `debug_assert!` for insert result will be useful to prevent logic bug.
- I'd like not to put side-effect in functional styled codes. I expect `unwrap_or_else` is immutable operation unless it panics. (though this is a sort of preference)